lib/sync/post.rb

```
@@ -0,0 +1,148 @@
+    attr_accessor :slug, :storage, :metadata, :body_markdown
+
+    def initialize(slug, storage: Post.storage)
+      @storage = storage
+      @slug = slug
+    end
+
+    def self.for(article, status, storage: Post.storage)
+      metadata = generate_metadata(article, status)
+      content = prepare_markdown(article["body_markdown"].to_s)
+
+      Post.new(status[:slug], storage:).tap do |post|
```

RuboCop (1.69.1)

[fatal] unexpected token tRPAREN
(Using Ruby 2.7 parser; configure using `TargetRubyVersion` parameter, under `AllCops`)

(Lint/Syntax)